In [1]:
# import pandas as pd
import numpy as np
import os
import graphlab as gl
from graphlab import SFrame
import pandas as pd


In [2]:
weather = SFrame.read_csv(os.path.join("data", "weather_filled_gl300_md10.csv"))
train = SFrame.read_csv(os.path.join("data", "train.csv"))
key = SFrame.read_csv(os.path.join("data", "key.csv"))
test = SFrame.read_csv(os.path.join("data", "test.csv"))


[INFO] Start server at: ipc:///tmp/graphlab_server-3636 - Server binary: /usr/local/lib/python2.7/dist-packages/graphlab/unity_server - Server log: /tmp/graphlab_server_1429990041.log
[INFO] GraphLab Server Version: 1.3.0


PROGRESS: Finished parsing file /media/vladimir/1ab2d5e6-a134-47e7-ba27-b2d70ac5ffc5/workspace/kaggle_wallmart/data/weather_filled_gl300_md10.csv
PROGRESS: Parsing completed. Parsed 100 lines in 0.133576 secs.
------------------------------------------------------
Inferred types from first line of file as 
column_type_hints=[int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,int,float,float,float,float,float,float,float,float,float,float,float,float,float,float,float,float,str]
If parsing fails due to incorrect types, you can correct
the inferred type list above and pass it to read_csv in
the column_type_hints argument
------------------------------------------------------
PROGRESS: Finished parsing file /media/vladimir/1ab2d5e6-a134-47e7-ba27-b2d70ac5ffc5/workspace/kaggle_wallmart/data/weather_filled_gl300_md10.csv
PROGRESS: Parsing completed. Parsed 20517 lines in 0.089886 secs.
PROGRESS: Finished parsing file /media/vl

In [3]:
weather_new = weather.join(key)

training = train.join(weather_new)
# training = training.join(weather)
testing = test.join(weather_new)
# testing = testing.join(weather)

In [4]:
print weather.num_rows()
print key.num_rows()
print training.num_rows()
print testing.num_rows()
print train.num_rows()
print test.num_rows()

20517
45
4617600
526917
4617600
526917


Is says that training is 4617600 lines and testing is 526917

In [5]:
print training.column_names()
training = training.sort(['date', 'store_nbr', 'item_nbr'])

['date', 'store_nbr', 'item_nbr', 'units', 'station_nbr', 'HZ', 'FU', 'UP', 'VCTS', 'DZ', 'BR', 'FG', 'BCFG', 'DU', 'FZRA', 'TS', 'RA', 'PL', 'GS', 'GR', 'FZDZ', 'VCFG', 'PRFG', 'FG+', 'TSRA', 'FZFG', 'BLDU', 'MIFG', 'SQ', 'BLSN', 'SN', 'SG', 'month', 'day', 'log_resultspeed', 'resultdir', 'dewpoint', 'log_preciptotal', 'log_avgspeed', 'tmax', 'tmin', 'stnpressure', 'wetbulb', 'cool', 'heat', 'tavg', 'sealevel', 'log_snowfall', 'day_length', 'depart']


In [6]:
training.head()

Columns:
	date	str
	store_nbr	int
	item_nbr	int
	units	int
	station_nbr	int
	HZ	int
	FU	int
	UP	int
	VCTS	int
	DZ	int
	BR	int
	FG	int
	BCFG	int
	DU	int
	FZRA	int
	TS	int
	RA	int
	PL	int
	GS	int
	GR	int
	FZDZ	int
	VCFG	int
	PRFG	int
	FG+	int
	TSRA	int
	FZFG	int
	BLDU	int
	MIFG	int
	SQ	int
	BLSN	int
	SN	int
	SG	int
	month	int
	day	int
	log_resultspeed	float
	resultdir	float
	dewpoint	float
	log_preciptotal	float
	log_avgspeed	float
	tmax	float
	tmin	float
	stnpressure	float
	wetbulb	float
	cool	float
	heat	float
	tavg	float
	sealevel	float
	log_snowfall	float
	day_length	float
	depart	float

Rows: 10

Data:
+------------+-----------+----------+-------+-------------+----+----+----+------+
|    date    | store_nbr | item_nbr | units | station_nbr | HZ | FU | UP | VCTS |
+------------+-----------+----------+-------+-------------+----+----+----+------+
| 2012-01-01 |     1     |    1     |   0   |      1      | 0  | 0  | 0  |  0   |
| 2012-01-01 |     1     |    2     |   0   |      1      | 0  | 0  | 0  |  0   |
| 2012-01-01 |     1     |    3     |   0   |      1      | 0  | 0  | 0  |  0   |
| 2012-01-01 |     1     |    4     |   0   |      1      | 0  | 0  | 0  |  0   |
| 2012-01-01 |     1     |    5     |   0   |      1      | 0  | 0  | 0  |  0   |
| 2012-01-01 |     1     |    6     |   0   |      1      | 0  | 0  | 0  |  0   |
| 2012-01-01 |     1     |    7     |   0   |      1      | 0  | 0  | 0  |  0   |
| 2012-01-01 |     1     |    8     |   0   |      1      | 0  | 0  | 0  |  0   |
| 2012-01-01 |     1     |    9     |   29  |      1      | 0  | 0  | 0  |  0   |
| 2012-01-01 |     1     |    10    |   0   |      1      | 0  | 0  | 0  |  0   |
+------------+-----------+----------+-------+-------------+----+----+----+------+
+----+----+----+------+----+------+----+----+----+----+----+-----+
| DZ | BR | FG | BCFG | DU | FZRA | TS | RA | PL | GS | GR | ... |
+----+----+----+------+----+------+----+----+----+----+----+-----+
| 0  | 1  | 1  |  0   | 0  |  0   | 0  | 1  | 0  | 0  | 0  | ... |
| 0  | 1  | 1  |  0   | 0  |  0   | 0  | 1  | 0  | 0  | 0  | ... |
| 0  | 1  | 1  |  0   | 0  |  0   | 0  | 1  | 0  | 0  | 0  | ... |
| 0  | 1  | 1  |  0   | 0  |  0   | 0  | 1  | 0  | 0  | 0  | ... |
| 0  | 1  | 1  |  0   | 0  |  0   | 0  | 1  | 0  | 0  | 0  | ... |
| 0  | 1  | 1  |  0   | 0  |  0   | 0  | 1  | 0  | 0  | 0  | ... |
| 0  | 1  | 1  |  0   | 0  |  0   | 0  | 1  | 0  | 0  | 0  | ... |
| 0  | 1  | 1  |  0   | 0  |  0   | 0  | 1  | 0  | 0  | 0  | ... |
| 0  | 1  | 1  |  0   | 0  |  0   | 0  | 1  | 0  | 0  | 0  | ... |
| 0  | 1  | 1  |  0   | 0  |  0   | 0  | 1  | 0  | 0  | 0  | ... |
+----+----+----+------+----+------+----+----+----+----+----+-----+
[10 rows x 50 columns]

In [7]:
print weather.num_rows()
print key.num_rows()

20517
45


In [8]:
print testing.num_rows()
print test.num_rows()

526917
526917


In [69]:
weather = weather.sort(['date', 'station_nbr'])

In [70]:
weather.head()

Columns:
	station_nbr	int
	tmax	float
	tmin	float
	tavg	float
	depart	float
	dewpoint	float
	wetbulb	float
	heat	float
	cool	float
	snowfall	float
	preciptotal	float
	stnpressure	float
	sealevel	float
	resultspeed	float
	resultdir	float
	avgspeed	float
	HZ	int
	FU	int
	UP	int
	VCTS	int
	DZ	int
	BR	int
	FG	int
	BCFG	int
	DU	int
	FZRA	int
	TS	int
	RA	int
	PL	int
	GS	int
	GR	int
	FZDZ	int
	VCFG	int
	PRFG	int
	FG+	int
	TSRA	int
	FZFG	int
	BLDU	int
	MIFG	int
	SQ	int
	BLSN	int
	SN	int
	SG	int
	month	int
	day	int
	day_length	float
	sunset_hour	float
	sunset_minute	float
	sunrise_hour	float
	sunrise_minute	float
	resultdirNAN	int
	resultspeedNAN	int
	dewpointNAN	int
	preciptotalNAN	int
	avgspeedNAN	int
	tmaxNAN	int
	tminNAN	int
	stnpressureNAN	int
	wetbulbNAN	int
	coolNAN	int
	heatNAN	int
	tavgNAN	int
	sealevelNAN	int
	snowfallNAN	int
	day_lengthNAN	int
	departNAN	int
	date	str

Rows: 10

Data:
+-------------+------+------+------+--------+----------+---------+------+------+
| station_nbr | tmax | tmin | tavg | depart | dewpoint | wetbulb | heat | cool |
+-------------+------+------+------+--------+----------+---------+------+------+
|      2      | 48.0 | 33.0 | 41.0 |  16.0  |   37.0   |   39.0  | 24.0 | 0.0  |
|      2      | 28.0 | 9.0  | 19.0 |  -6.0  |   4.0    |   16.0  | 46.0 | 0.0  |
|      2      | 46.0 | 28.0 | 37.0 |  12.0  |   24.0   |   32.0  | 28.0 | 0.0  |
|      3      | 55.0 | 34.0 | 45.0 |  9.0   |   24.0   |   36.0  | 20.0 | 0.0  |
|      3      | 45.0 | 24.0 | 35.0 |  -1.0  |   11.0   |   28.0  | 30.0 | 0.0  |
|      4      | 63.0 | 47.0 | 55.0 |  4.0   |   28.0   |   43.0  | 10.0 | 0.0  |
|      4      | 60.0 | 37.0 | 49.0 |  -2.0  |   26.0   |   39.0  | 16.0 | 0.0  |
|      6      | 63.0 | 34.0 | 49.0 |  0.0   |   31.0   |   43.0  | 16.0 | 0.0  |
|      6      | 60.0 | 30.0 | 45.0 |  -4.0  |   27.0   |   38.0  | 20.0 | 0.0  |
|      11     | 72.0 | 48.0 | 60.0 |  7.0   |   54.0   |   56.0  | 5.0  | 0.0  |
+-------------+------+------+------+--------+----------+---------+------+------+
+----------+-------------+-------------+----------+-------------+-----------+
| snowfall | preciptotal | stnpressure | sealevel | resultspeed | resultdir |
+----------+-------------+-------------+----------+-------------+-----------+
|   0.0    |     0.07    |    28.82    |  29.91   |     9.1     |    23.0   |
|   0.0    |     0.0     |    28.69    |  29.85   |     12.3    |    31.0   |
|   0.01   |     0.01    |    28.51    |  29.62   |     12.7    |    26.0   |
|   0.0    |     0.0     |    29.77    |  30.47   |     9.9     |    31.0   |
|   0.0    |     0.0     |    29.89    |  30.63   |     7.7     |    32.0   |
|   0.0    |     0.0     |    29.79    |  30.48   |     8.0     |    35.0   |
|   0.0    |     0.0     |    29.93    |  30.64   |     3.7     |    36.0   |
|   0.0    |     0.0     |    29.95    |  30.47   |     14.0    |    36.0   |
|   0.0    |     0.0     |    30.09    |  30.63   |     9.0     |    1.0    |
|   0.0    |     0.0     |    30.15    |  30.18   |     4.6     |    23.0   |
+----------+-------------+-------------+----------+-------------+-----------+
+----------+----+----+----+------+-----+
| avgspeed | HZ | FU | UP | VCTS | ... |
+----------+----+----+----+------+-----+
|   11.3   | 0  | 0  | 0  |  0   | ... |
|   12.8   | 0  | 0  | 0  |  0   | ... |
|   13.3   | 0  | 0  | 0  |  0   | ... |
|   10.0   | 0  | 0  | 0  |  0   | ... |
|   8.2    | 0  | 0  | 0  |  0   | ... |
|   8.2    | 0  | 0  | 0  |  0   | ... |
|   4.8    | 0  | 0  | 0  |  0   | ... |
|   13.8   | 0  | 0  | 0  |  0   | ... |
|   9.3    | 0  | 0  | 0  |  0   | ... |
|   4.8    | 0  | 0  | 0  |  0   | ... |
+----------+----+----+----+------+-----+
[10 rows x 67 columns]

In [49]:
key.head()

Columns:
	store_nbr	int
	station_nbr	int
	tmax	float
	tmin	float
	tavg	float
	depart	float
	dewpoint	float
	wetbulb	float
	heat	float
	cool	float
	snowfall	float
	preciptotal	float
	stnpressure	float
	sealevel	float
	resultspeed	float
	resultdir	float
	avgspeed	float
	HZ	int
	FU	int
	UP	int
	VCTS	int
	DZ	int
	BR	int
	FG	int
	BCFG	int
	DU	int
	FZRA	int
	TS	int
	RA	int
	PL	int
	GS	int
	GR	int
	FZDZ	int
	VCFG	int
	PRFG	int
	FG+	int
	TSRA	int
	FZFG	int
	BLDU	int
	MIFG	int
	SQ	int
	BLSN	int
	SN	int
	SG	int
	month	int
	day	int
	day_length	float
	sunset_hour	float
	sunset_minute	float
	sunrise_hour	float
	sunrise_minute	float
	resultdirNAN	int
	resultspeedNAN	int
	dewpointNAN	int
	preciptotalNAN	int
	avgspeedNAN	int
	tmaxNAN	int
	tminNAN	int
	stnpressureNAN	int
	wetbulbNAN	int
	coolNAN	int
	heatNAN	int
	tavgNAN	int
	sealevelNAN	int
	snowfallNAN	int
	day_lengthNAN	int
	departNAN	int
	date	str

Rows: 10

Data:
+-----------+-------------+------+------+------+--------+----------+---------+
| store_nbr | station_nbr | tmax | tmin | tavg | depart | dewpoint | wetbulb |
+-----------+-------------+------+------+------+--------+----------+---------+
|     16    |      2      | 48.0 | 33.0 | 41.0 |  16.0  |   37.0   |   39.0  |
|     21    |      3      | 55.0 | 34.0 | 45.0 |  9.0   |   24.0   |   36.0  |
|     29    |      3      | 55.0 | 34.0 | 45.0 |  9.0   |   24.0   |   36.0  |
|     33    |      3      | 55.0 | 34.0 | 45.0 |  9.0   |   24.0   |   36.0  |
|     8     |      4      | 63.0 | 47.0 | 55.0 |  4.0   |   28.0   |   43.0  |
|     7     |      6      | 63.0 | 34.0 | 49.0 |  0.0   |   31.0   |   43.0  |
|     13    |      6      | 63.0 | 34.0 | 49.0 |  0.0   |   31.0   |   43.0  |
|     12    |      11     | 72.0 | 48.0 | 60.0 |  7.0   |   54.0   |   56.0  |
|     43    |      11     | 72.0 | 48.0 | 60.0 |  7.0   |   54.0   |   56.0  |
|     2     |      14     | 50.0 | 34.0 | 42.0 |  5.0   |   25.0   |   35.0  |
+-----------+-------------+------+------+------+--------+----------+---------+
+------+------+----------+-------------+-------------+----------+-------------+
| heat | cool | snowfall | preciptotal | stnpressure | sealevel | resultspeed |
+------+------+----------+-------------+-------------+----------+-------------+
| 24.0 | 0.0  |   0.0    |     0.07    |    28.82    |  29.91   |     9.1     |
| 20.0 | 0.0  |   0.0    |     0.0     |    29.77    |  30.47   |     9.9     |
| 20.0 | 0.0  |   0.0    |     0.0     |    29.77    |  30.47   |     9.9     |
| 20.0 | 0.0  |   0.0    |     0.0     |    29.77    |  30.47   |     9.9     |
| 10.0 | 0.0  |   0.0    |     0.0     |    29.79    |  30.48   |     8.0     |
| 16.0 | 0.0  |   0.0    |     0.0     |    29.95    |  30.47   |     14.0    |
| 16.0 | 0.0  |   0.0    |     0.0     |    29.95    |  30.47   |     14.0    |
| 5.0  | 0.0  |   0.0    |     0.0     |    30.15    |  30.18   |     4.6     |
| 5.0  | 0.0  |   0.0    |     0.0     |    30.15    |  30.18   |     4.6     |
| 23.0 | 0.0  |   0.0    |     0.0     |    29.13    |  30.52   |     11.4    |
+------+------+----------+-------------+-------------+----------+-------------+
+-----------+----------+----+----+----+-----+
| resultdir | avgspeed | HZ | FU | UP | ... |
+-----------+----------+----+----+----+-----+
|    23.0   |   11.3   | 0  | 0  | 0  | ... |
|    31.0   |   10.0   | 0  | 0  | 0  | ... |
|    31.0   |   10.0   | 0  | 0  | 0  | ... |
|    31.0   |   10.0   | 0  | 0  | 0  | ... |
|    35.0   |   8.2    | 0  | 0  | 0  | ... |
|    36.0   |   13.8   | 0  | 0  | 0  | ... |
|    36.0   |   13.8   | 0  | 0  | 0  | ... |
|    23.0   |   4.8    | 0  | 0  | 0  | ... |
|    23.0   |   4.8    | 0  | 0  | 0  | ... |
|    32.0   |   11.3   | 0  | 0  | 0  | ... |
+-----------+----------+----+----+----+-----+
[10 rows x 68 columns]

In [42]:
train.head()

Columns:
	date	str
	store_nbr	int
	item_nbr	int
	units	int

Rows: 10

Data:
+------------+-----------+----------+-------+
|    date    | store_nbr | item_nbr | units |
+------------+-----------+----------+-------+
| 2012-01-01 |     1     |    1     |   0   |
| 2012-01-01 |     1     |    2     |   0   |
| 2012-01-01 |     1     |    3     |   0   |
| 2012-01-01 |     1     |    4     |   0   |
| 2012-01-01 |     1     |    5     |   0   |
| 2012-01-01 |     1     |    6     |   0   |
| 2012-01-01 |     1     |    7     |   0   |
| 2012-01-01 |     1     |    8     |   0   |
| 2012-01-01 |     1     |    9     |   29  |
| 2012-01-01 |     1     |    10    |   0   |
+------------+-----------+----------+-------+
[10 rows x 4 columns]

In [15]:
training.column_names()

['date',
 'store_nbr',
 'item_nbr',
 'units',
 'station_nbr',
 'HZ',
 'FU',
 'UP',
 'VCTS',
 'DZ',
 'BR',
 'FG',
 'BCFG',
 'DU',
 'FZRA',
 'TS',
 'RA',
 'PL',
 'GS',
 'GR',
 'FZDZ',
 'VCFG',
 'PRFG',
 'FG+',
 'TSRA',
 'FZFG',
 'BLDU',
 'MIFG',
 'SQ',
 'BLSN',
 'SN',
 'SG',
 'month',
 'day',
 'log_resultspeed',
 'resultdir',
 'dewpoint',
 'log_preciptotal',
 'log_avgspeed',
 'tmax',
 'tmin',
 'stnpressure',
 'wetbulb',
 'cool',
 'heat',
 'tavg',
 'sealevel',
 'log_snowfall',
 'day_length',
 'depart']

In [5]:
features =['store_nbr',
 'item_nbr',
#  'units',
 'station_nbr',
 'tmax',
 'tmin',
 'tavg',
 'depart',
 'dewpoint',
 'wetbulb',
 'heat',
 'cool',
 'log_snowfall',
 'log_preciptotal',
 'stnpressure',
 'sealevel',
 'log_resultspeed',
 'resultdir',
 'log_avgspeed',
 'HZ',
 'FU',
 'UP',
 'VCTS',
 'DZ',
 'BR',
 'FG',
 'BCFG',
 'DU',
 'FZRA',
 'TS',
 'RA',
 'PL',
 'GS',
 'GR',
 'FZDZ',
 'VCFG',
 'PRFG',
 'FG+',
 'TSRA',
 'FZFG',
 'BLDU',
 'MIFG',
 'SQ',
 'BLSN',
 'SN',
 'SG',
#  'month',
#  'day',
 'day_length',
#  'sunset_hour',
#  'sunset_minute',
#  'sunrise_hour',
#  'sunrise_minute',
#  'resultdirNAN',
#  'resultspeedNAN',
#  'dewpointNAN',
#  'preciptotalNAN',
#  'avgspeedNAN',
#  'tmaxNAN',
#  'tminNAN',
#  'stnpressureNAN',
#  'wetbulbNAN',
#  'coolNAN',
#  'heatNAN',
#  'tavgNAN',
#  'sealevelNAN',
#  'snowfallNAN',
#  'day_lengthNAN',
#  'departNAN'
          ]

In [6]:
sf_train, sf_test = training.random_split(0.8, seed=5)

In [ ]:
# from gl.boosted_trees_regression import BoostedTreesRegression

# model = gl.boosted_trees_regression.create(sf_train, 
#                              target='units', 
#                              features=features,
#                              max_depth=20,
#                              step_size=0.1,
#                              max_iterations=100,
#                              validation_set = sf_test)

model = gl.boosted_trees_regression.create(sf_train, 
                             target='units', 
                             features=features,                             
                             validation_set = sf_test,
                                          max_iterations=50)
# model = gl.boosted_trees_regression.create(sf_train, target='units', 
#                                            features=features, 
#                                            max_iterations=50, 
#                                            max_depth=10,
# #                                            row_subsample=0.8,
#                                             step_size=0.1,
#                                            validation_set=sf_test)

In [9]:
prediction_testing = model.predict(testing)

In [10]:
len(prediction_testing)

526917

In [54]:
testing.num_rows()

508491

In [46]:
prediction_training = model.predict(training)

In [26]:
prediction = prediction_testing

In [11]:
pd_testing = pd.DataFrame()
pd_testing["units"] = prediction_testing

# pd_testing.to_csv(os.path.join("predictions", "xgbt_default_testing.csv"))

In [52]:
pd_training = pd.DataFrame()
pd_training["units"] = prediction_training

pd_training.to_csv(os.path.join("predictions", "xgbt_md20_ss01_mi100_training.csv"))

In [48]:
prediction_training.save(os.path.join("predictions", "xgbt_md20_ss01_mi100_training.csv"), format='csv')

In [20]:
submission = pd.DataFrame()

In [21]:
def merge_data(df):
    return ''.join([str(df["store_nbr"]), "_", str(df["item_nbr"]), "_", df["date"]])

In [22]:
submission["id"] = testing[["store_nbr", "item_nbr", "date"]].apply(merge_data)

In [36]:
testing.head()

Columns:
	date	str
	store_nbr	int
	item_nbr	int
	station_nbr	int
	tmax	float
	tmin	float
	tavg	float
	depart	float
	dewpoint	float
	wetbulb	float
	heat	float
	cool	float
	snowfall	float
	preciptotal	float
	stnpressure	float
	sealevel	float
	resultspeed	float
	resultdir	float
	avgspeed	float
	HZ	int
	FU	int
	UP	int
	VCTS	int
	DZ	int
	BR	int
	FG	int
	BCFG	int
	DU	int
	FZRA	int
	TS	int
	RA	int
	PL	int
	GS	int
	GR	int
	FZDZ	int
	VCFG	int
	PRFG	int
	FG+	int
	TSRA	int
	FZFG	int
	BLDU	int
	MIFG	int
	SQ	int
	BLSN	int
	SN	int
	SG	int
	month	int
	day	int
	day_length	float
	sunset_hour	float
	sunset_minute	float
	sunrise_hour	float
	sunrise_minute	float
	resultdirNAN	int
	resultspeedNAN	int
	dewpointNAN	int
	preciptotalNAN	int
	avgspeedNAN	int
	tmaxNAN	int
	tminNAN	int
	stnpressureNAN	int
	wetbulbNAN	int
	coolNAN	int
	heatNAN	int
	tavgNAN	int
	sealevelNAN	int
	snowfallNAN	int
	day_lengthNAN	int
	departNAN	int

Rows: 10

Data:
+------------+-----------+----------+-------------+------+------+------+--------+
|    date    | store_nbr | item_nbr | station_nbr | tmax | tmin | tavg | depart |
+------------+-----------+----------+-------------+------+------+------+--------+
| 2013-04-01 |     2     |    1     |      14     | 95.0 | 71.0 | 83.0 |  8.0   |
| 2013-04-01 |     2     |    1     |      14     | 88.0 | 62.0 | 75.0 |  -6.0  |
| 2013-04-01 |     2     |    2     |      14     | 95.0 | 71.0 | 83.0 |  8.0   |
| 2013-04-01 |     2     |    2     |      14     | 88.0 | 62.0 | 75.0 |  -6.0  |
| 2013-04-01 |     2     |    3     |      14     | 95.0 | 71.0 | 83.0 |  8.0   |
| 2013-04-01 |     2     |    3     |      14     | 88.0 | 62.0 | 75.0 |  -6.0  |
| 2013-04-01 |     2     |    4     |      14     | 95.0 | 71.0 | 83.0 |  8.0   |
| 2013-04-01 |     2     |    4     |      14     | 88.0 | 62.0 | 75.0 |  -6.0  |
| 2013-04-01 |     2     |    5     |      14     | 95.0 | 71.0 | 83.0 |  8.0   |
| 2013-04-01 |     2     |    5     |      14     | 88.0 | 62.0 | 75.0 |  -6.0  |
+------------+-----------+----------+-------------+------+------+------+--------+
+---------------+---------------+------+------+----------+-------------+
|    dewpoint   |    wetbulb    | heat | cool | snowfall | preciptotal |
+---------------+---------------+------+------+----------+-------------+
|      67.0     | 70.0115029973 | 0.0  | 18.0 |   0.0    |     0.0     |
| 63.5360749548 | 65.8983139198 | 0.0  | 10.0 |   0.0    |     0.0     |
|      67.0     | 70.0115029973 | 0.0  | 18.0 |   0.0    |     0.0     |
| 63.5360749548 | 65.8983139198 | 0.0  | 10.0 |   0.0    |     0.0     |
|      67.0     | 70.0115029973 | 0.0  | 18.0 |   0.0    |     0.0     |
| 63.5360749548 | 65.8983139198 | 0.0  | 10.0 |   0.0    |     0.0     |
|      67.0     | 70.0115029973 | 0.0  | 18.0 |   0.0    |     0.0     |
| 63.5360749548 | 65.8983139198 | 0.0  | 10.0 |   0.0    |     0.0     |
|      67.0     | 70.0115029973 | 0.0  | 18.0 |   0.0    |     0.0     |
| 63.5360749548 | 65.8983139198 | 0.0  | 10.0 |   0.0    |     0.0     |
+---------------+---------------+------+------+----------+-------------+
+---------------+---------------+---------------+---------------+----------+----+-----+
|  stnpressure  |    sealevel   |  resultspeed  |   resultdir   | avgspeed | HZ | ... |
+---------------+---------------+---------------+---------------+----------+----+-----+
| 28.1864547022 | 29.0200885972 |      15.2     |      17.0     |   15.0   | 0  | ... |
| 28.5570991943 | 29.1003321089 | 7.24105173768 | 16.5184229047 |   6.8    | 0  | ... |
| 28.1864547022 | 29.0200885972 |      15.2     |      17.0     |   15.0   | 0  | ... |
| 28.5570991943 | 29.1003321089 | 7.24105173768 | 16.5184229047 |   6.8    | 0  | ... |
| 28.1864547022 | 29.0200885972 |      15.2     |      17.0     |   15.0   | 0  | ... |
| 28.5570991943 | 29.1003321089 | 7.24105173768 | 16.5184229047 |   6.8    | 0  | ... |
| 28.1864547022 | 29.0200885972 |      15.2     |      17.0     |   15.0   | 0  | ... |
| 28.5570991943 | 29.

In [23]:
# import pandas as pd
result = submission

In [24]:
result["units"] = prediction_testing
# result["units"] = result["units"].apply(round).astype(int)
result["units"] = result["units"].apply(int).astype(int)


In [25]:
result["units"][result["units"] < 0]

1265     -2
2597     -2
3929     -2
4151     -1
4262     -1
5261     -2
5483     -1
5594     -1
6593     -2
7703     -2
11698    -3
11699    -1
11809    -3
11810    -1
11920    -2
12031    -3
12032    -1
12142    -3
12143    -1
12253    -2
12254    -2
12364    -2
12365    -2
13363    -3
13364    -1
13474    -3
13475    -1
13585    -2
13696    -3
13697    -1
         ..
524186   -4
524259   -1
524407   -2
524518   -2
524629   -1
524703   -1
524722   -1
524723   -1
524741   -4
524764   -1
524814   -1
525073   -2
525184   -1
525258   -1
525277   -1
525278   -1
525319   -1
525369   -1
525628   -2
525739   -1
525813   -1
525832   -1
525833   -1
525874   -1
525924   -1
526368   -1
526387   -1
526388   -1
526429   -1
526479   -1
Name: units, dtype: int64

In [27]:
result[result["units"] < 0]['units'] = 0

In [28]:
#save submission
result.to_csv(os.path.join("predictions", "Graphlab_default.csv"), index=False)

In [42]:
import pandas as pd
result = submission

In [43]:
result["units"][result["units"] < 0] = 0
result.to_csv(os.path.join("predictions", "Graphlab_default_rf.csv"), index=False)

In [44]:
features_1 = ['item_nbr', 'HZ', 'FU', 'UP', 'TSSN', 'VCTS', 'DZ', 'BR', 'FG',
 'BCFG', 'DU', 'FZRA', 'TS', 'RA', 'PL', 'GS', 'GR', 'FZDZ', 'VCFG', 'PRFG', 'FG+', 'TSRA', 'FZFG', 'BLDU', 'MIFG', 'SQ',
 'BLSN', 'SN', 'SG', 'month', 'day']

In [45]:
model = gl.regression.create(training, target='units', features=features_1)

PROGRESS: Boosted trees regression:

PROGRESS: --------------------------------------------------------

PROGRESS: Number of examples          : 4617600

PROGRESS: Number of features          : 31

PROGRESS: Number of unpacked features : 31

PROGRESS: Starting Boosted Trees

PROGRESS: --------------------------------------------------------

PROGRESS:   Iter        RMSE Elapsed time

PROGRESS:      0   9.480e+00       16.16s

PROGRESS:      1   9.272e+00       24.93s

PROGRESS:      2   9.166e+00       32.83s

PROGRESS:      3   9.110e+00       40.49s

PROGRESS:      4   9.082e+00       48.70s

PROGRESS:      5   9.062e+00       58.98s

PROGRESS:      6   9.053e+00       67.51s

PROGRESS:      7   9.045e+00       78.32s

PROGRESS:      8   9.042e+00       86.39s

PROGRESS:      9   9.038e+00       94.20s

In [46]:
features_2 = ['item_nbr', 'tmax', 'tmin', 'tavg', 'depart', 'dewpoint', 'wetbulb', 'heat', 'cool', 'snowfall', 'preciptotal',
 'stnpressure', 'sealevel', 'resultspeed', 'resultdir', 'avgspeed', 'month', 'day', 'day_length']

In [47]:
model = gl.regression.create(training, target='units', features=features_2)

PROGRESS: Boosted trees regression:

PROGRESS: --------------------------------------------------------

PROGRESS: Number of examples          : 4617600

PROGRESS: Number of features          : 19

PROGRESS: Number of unpacked features : 19

PROGRESS: Starting Boosted Trees

PROGRESS: --------------------------------------------------------

PROGRESS:   Iter        RMSE Elapsed time

PROGRESS:      0   9.305e+00       11.55s

PROGRESS:      1   8.997e+00       17.71s

PROGRESS:      2   8.825e+00       23.84s

PROGRESS:      3   8.733e+00       30.45s

PROGRESS:      4   8.677e+00       37.75s

PROGRESS:      5   8.646e+00       43.96s

PROGRESS:      6   8.621e+00       50.11s

PROGRESS:      7   8.599e+00       58.21s

PROGRESS:      8   8.582e+00       64.91s

PROGRESS:      9   8.572e+00       71.64s

In [49]:
features_3 = ['item_nbr', 'tmax', 'tmin', 'tavg', 'depart', 'dewpoint', 'wetbulb', 'heat', 'cool', 'snowfall', 'preciptotal',
 'stnpressure', 'sealevel', 'resultspeed', 'resultdir', 'avgspeed', 'HZ', 'FU', 'TSSN', 'VCTS', 'DZ', 'BR', 'FG',
 'BCFG', 'DU', 'FZRA', 'TS', 'RA', 'PL', 'GS', 'GR', 'FZDZ', 'VCFG', 'PRFG', 'FG+', 'TSRA', 'FZFG', 'BLDU', 'MIFG', 'SQ',
 'BLSN', 'SN', 'SG', 'day_length']

In [50]:
model = gl.regression.create(training, target='units', features=features_3)

PROGRESS: Boosted trees regression:

PROGRESS: --------------------------------------------------------

PROGRESS: Number of examples          : 4617600

PROGRESS: Number of features          : 44

PROGRESS: Number of unpacked features : 44

PROGRESS: Starting Boosted Trees

PROGRESS: --------------------------------------------------------

PROGRESS:   Iter        RMSE Elapsed time

PROGRESS:      0   9.301e+00       20.56s

PROGRESS:      1   8.998e+00       30.46s

PROGRESS:      2   8.825e+00       41.36s

PROGRESS:      3   8.733e+00       53.05s

PROGRESS:      4   8.675e+00       65.89s

PROGRESS:      5   8.643e+00       76.39s

PROGRESS:      6   8.621e+00       89.42s

PROGRESS:      7   8.597e+00      103.31s

PROGRESS:      8   8.587e+00      117.16s

PROGRESS:      9   8.559e+00      132.75s

In [56]:
model = gl.boosted_trees_regression.create(training, target='units', features=features, max_iterations=100)

PROGRESS: Boosted trees regression:

PROGRESS: --------------------------------------------------------

PROGRESS: Number of examples          : 4617600

PROGRESS: Number of features          : 46

PROGRESS: Number of unpacked features : 46

PROGRESS: Starting Boosted Trees

PROGRESS: --------------------------------------------------------

PROGRESS:   Iter        RMSE Elapsed time

PROGRESS:      0   9.305e+00       22.79s

PROGRESS:      1   8.997e+00       33.77s

PROGRESS:      2   8.825e+00       46.29s

PROGRESS:      3   8.734e+00       58.25s

PROGRESS:      4   8.678e+00       71.45s

PROGRESS:      5   8.646e+00       85.19s

PROGRESS:      6   8.620e+00       96.52s

PROGRESS:      7   8.595e+00      109.55s

PROGRESS:      8   8.580e+00      122.09s

PROGRESS:      9   8.564e+00      132.97s

PROGRESS:     10   8.548e+00      145.13s

PROGRESS:     11   8.535e+00      156.41s

PROGRESS:     12   8.528e+00      168.42s

PROGRESS:     13   8.520e+00      179.76s

PROGRESS:     14   8.514e+00      191.93s

PROGRESS:     15   8.508e+00      204.70s

PROGRESS:     16   8.501e+00      218.55s

PROGRESS:     17   8.497e+00      232.81s

PROGRESS:     18   8.486e+00      246.35s

PROGRESS:     19   8.480e+00      259.32s

PROGRESS:     20   8.477e+00      273.03s

PROGRESS:     21   8.449e+00      287.95s

PROGRESS:     22   8.445e+00      301.10s

PROGRESS:     23   8.441e+00      314.17s

PROGRESS:     24   8.435e+00      327.80s

PROGRESS:     25   8.433e+00      343.70s

PROGRESS:     26   8.420e+00      357.40s

PROGRESS:     27   8.416e+00      372.23s

PROGRESS:     28   8.413e+00      388.93s

PROGRESS:     29   8.396e+00      404.39s

PROGRESS:     30   8.394e+00      418.30s

PROGRESS:     31   8.387e+00      432.92s

PROGRESS:     32   8.376e+00      450.38s

PROGRESS:     33   8.369e+00      469.84s

PROGRESS:     34   8.365e+00      486.44s

PROGRESS:     35   8.362e+00      506.83s

PROGRESS:     36   8.360e+00      526.52s

PROGRESS:     37   8.358e+00      546.41s

PROGRESS:     38   8.352e+00      563.88s

PROGRESS:     39   8.349e+00      580.75s

PROGRESS:     40   8.344e+00      597.48s

PROGRESS:     41   8.336e+00      612.78s

PROGRESS:     42   8.332e+00      623.11s

PROGRESS:     43   8.330e+00      634.24s

PROGRESS:     44   8.318e+00      648.35s

PROGRESS:     45   8.317e+00      665.15s

PROGRESS:     46   8.317e+00      687.86s

PROGRESS:     47   8.314e+00      709.28s

PROGRESS:     48   8.309e+00      730.22s

PROGRESS:     49   8.306e+00      750.02s

PROGRESS:     50   8.302e+00      765.98s

PROGRESS:     51   8.297e+00      781.21s

PROGRESS:     52   8.295e+00      792.21s

PROGRESS:     53   8.292e+00      804.32s

PROGRESS:     54   8.288e+00      817.07s

PROGRESS:     55   8.284e+00      831.00s

PROGRESS:     56   8.268e+00      846.25s

PROGRESS:     57   8.264e+00      863.67s

PROGRESS:     58   8.262e+00      881.24s

PROGRESS:     59   8.259e+00      900.58s

PROGRESS:     60   8.258e+00      915.81s

PROGRESS:     61   8.255e+00      931.10s

PROGRESS:     62   8.252e+00      942.48s

PROGRESS:     63   8.248e+00      952.68s

PROGRESS:     64   8.244e+00      964.81s

PROGRESS:     65   8.240e+00      975.74s

PROGRESS:     66   8.238e+00      987.60s

PROGRESS:     67   8.231e+00      998.81s

PROGRESS:     68   8.229e+00     1010.94s

PROGRESS:     69   8.227e+00     1021.31s

PROGRESS:     70   8.222e+00     1031.99s

PROGRESS:     71   8.222e+00     1042.69s

PROGRESS:     72   8.217e+00     1054.87s

PROGRESS:     73   8.212e+00     1066.99s

PROGRESS:     74   8.208e+00     1083.25s

PROGRESS:     75   8.201e+00     1096.07s

PROGRESS:     76   8.186e+00     1111.97s

PROGRESS:     77   8.168e+00     1130.48s

PROGRESS:     78   8.164e+00     1148.39s

PROGRESS:     79   8.161e+00     1165.56s

PROGRESS:     80   8.150e+00     1185.01s

PROGRESS:     81   8.145e+00     1202.57s

PROGRESS:     82   8.144e+00     1217.62s

PROGRESS:     83   8.142e+00     1228.74s

PROGRESS:     84   8.142e+00     1243.17s

PROGRESS:     85   8.141e+00     1254.72s

PROGRESS:     86   8.138e+00     1267.51s

PROGRESS:     87   8.133e+00     1286.63s

PROGRESS:     88   8.128e+00     1303.28s

PROGRESS:     89   8.125e+00     1319.45s

PROGRESS:     90   8.123e+00     1337.80s

PROGRESS:     91   8.121e+00     1357.93s

PROGRESS:     92   8.120e+00     1378.46s

PROGRESS:     93   8.116e+00     1397.42s

PROGRESS:     94   8.112e+00     1410.35s

PROGRESS:     95   8.110e+00     1422.94s

PROGRESS:     96   8.106e+00     1436.76s

PROGRESS:     97   8.075e+00     1450.97s

PROGRESS:     98   8.073e+00     1464.78s

PROGRESS:     99   8.071e+00     1482.56s

In [71]:
model = gl.boosted_trees_regression.create(training, target='units', features=features, max_iterations=100)
print model.summary()

PROGRESS: Boosted trees regression:

PROGRESS: --------------------------------------------------------

PROGRESS: Number of examples          : 4617600

PROGRESS: Number of features          : 45

PROGRESS: Number of unpacked features : 45

PROGRESS: Starting Boosted Trees

PROGRESS: --------------------------------------------------------

PROGRESS:   Iter        RMSE Elapsed time

PROGRESS:      0   9.302e+00       21.78s

PROGRESS:      1   8.992e+00       34.04s

PROGRESS:      2   8.819e+00       46.21s

PROGRESS:      3   8.727e+00       57.59s

PROGRESS:      4   8.680e+00       69.98s

PROGRESS:      5   8.640e+00       84.88s

PROGRESS:      6   8.616e+00       98.06s

PROGRESS:      7   8.602e+00      110.74s

PROGRESS:      8   8.587e+00      122.21s

PROGRESS:      9   8.578e+00      136.12s

PROGRESS:     10   8.564e+00      148.71s

PROGRESS:     11   8.556e+00      161.30s

PROGRESS:     12   8.552e+00      174.72s

PROGRESS:     13   8.516e+00      188.41s

PROGRESS:     14   8.506e+00      198.59s

PROGRESS:     15   8.500e+00      210.61s

PROGRESS:     16   8.484e+00      221.76s

PROGRESS:     17   8.480e+00      234.19s

PROGRESS:     18   8.471e+00      244.35s

PROGRESS:     19   8.468e+00      256.70s

PROGRESS:     20   8.409e+00      268.98s

PROGRESS:     21   8.394e+00      280.46s

PROGRESS:     22   8.389e+00      293.28s

PROGRESS:     23   8.385e+00      305.93s

PROGRESS:     24   8.381e+00      318.86s

PROGRESS:     25   8.375e+00      330.12s

PROGRESS:     26   8.371e+00      342.34s

PROGRESS:     27   8.368e+00      355.01s

PROGRESS:     28   8.367e+00      367.30s

PROGRESS:     29   8.359e+00      378.71s

PROGRESS:     30   8.356e+00      390.20s

PROGRESS:     31   8.351e+00      404.10s

PROGRESS:     32   8.345e+00      417.21s

PROGRESS:     33   8.340e+00      429.86s

PROGRESS:     34   8.329e+00      441.54s

PROGRESS:     35   8.325e+00      454.59s

PROGRESS:     36   8.314e+00      467.67s

PROGRESS:     37   8.309e+00      480.97s

PROGRESS:     38   8.307e+00      493.22s

PROGRESS:     39   8.297e+00      504.46s

PROGRESS:     40   8.292e+00      517.35s

PROGRESS:     41   8.289e+00      529.85s

PROGRESS:     42   8.288e+00      541.34s

PROGRESS:     43   8.286e+00      553.01s

PROGRESS:     44   8.284e+00      565.15s

PROGRESS:     45   8.283e+00      576.36s

PROGRESS:     46   8.278e+00      587.87s

PROGRESS:     47   8.274e+00      599.99s

PROGRESS:     48   8.266e+00      610.88s

PROGRESS:     49   8.263e+00      623.22s

PROGRESS:     50   8.257e+00      635.02s

PROGRESS:     51   8.254e+00      647.52s

PROGRESS:     52   8.239e+00      658.01s

PROGRESS:     53   8.234e+00      670.29s

PROGRESS:     54   8.231e+00      682.26s

PROGRESS:     55   8.229e+00      693.37s

PROGRESS:     56   8.225e+00      703.84s

PROGRESS:     57   8.225e+00      715.88s

PROGRESS:     58   8.218e+00      728.04s

PROGRESS:     59   8.213e+00      738.87s

PROGRESS:     60   8.204e+00      752.43s

PROGRESS:     61   8.202e+00      764.91s

PROGRESS:     62   8.200e+00      777.03s

PROGRESS:     63   8.192e+00      789.51s

PROGRESS:     64   8.181e+00      801.87s

PROGRESS:     65   8.176e+00      814.21s

PROGRESS:     66   8.174e+00      825.15s

PROGRESS:     67   8.166e+00      837.13s

PROGRESS:     68   8.161e+00      849.63s

PROGRESS:     69   8.157e+00      861.49s

PROGRESS:     70   8.150e+00      873.42s

PROGRESS:     71   8.148e+00      885.39s

PROGRESS:     72   8.147e+00      897.76s

PROGRESS:     73   8.147e+00      910.51s

PROGRESS:     74   8.145e+00      922.68s

PROGRESS:     75   8.143e+00      934.21s

PROGRESS:     76   8.142e+00      947.09s

PROGRESS:     77   8.138e+00      959.71s

PROGRESS:     78   8.135e+00      971.47s

PROGRESS:     79   8.133e+00      984.23s

PROGRESS:     80   8.131e+00      997.23s

PROGRESS:     81   8.130e+00     1010.36s

PROGRESS:     82   8.128e+00     1022.80s

PROGRESS:     83   8.127e+00     1034.23s

PROGRESS:     84   8.126e+00     1046.34s

PROGRESS:     85   8.125e+00     1058.32s

PROGRESS:     86   8.123e+00     1070.30s

PROGRESS:     87   8.118e+00     1081.41s

PROGRESS:     88   8.117e+00     1092.51s

PROGRESS:     89   8.115e+00     1103.92s

PROGRESS:     90   8.114e+00     1115.66s

PROGRESS:     91   8.099e+00     1127.33s

PROGRESS:     92   8.096e+00     1139.25s

PROGRESS:     93   8.094e+00     1149.74s

PROGRESS:     94   8.091e+00     1162.00s

PROGRESS:     95   8.090e+00     1173.85s

PROGRESS:     96   8.087e+00     1185.25s

PROGRESS:     97   8.073e+00     1197.34s

PROGRESS:     98   8.069e+00     1209.18s

PROGRESS:     99   8.066e+00     1220.98s


                    Model summary                       
--------------------------------------------------------
Class                         : BoostedTreesRegression

Number of examples            : 4617600
Number of feature columns     : 45
Number of unpacked features   : 45

Number of trees               : 100
Max tree depth                : 6
Train RMSE                    : 8.0664
Validation RMSE               : None
Training time (sec)           : 1224.6736

None


In [72]:
prediction = model.predict(testing)

In [73]:
submission = pd.DataFrame()

In [74]:
submission["id"] = testing[["store_nbr", "item_nbr", "date"]].apply(merge_data)

In [31]:
submission["units"] = prediction

In [32]:
import pandas as pd
result = submission

In [34]:
result["units"][result["units"] < 0] = 0
result["units"] = result["units"].apply(round).astype(int)
result.to_csv(os.path.join("predictions", "Graphlab_100_gl300.csv"), index=False)